include/nlohmann/detail/output/serializer.hpp

```diff
@@ -389,9 +386,9 @@ class serializer
 
                 case UTF8_REJECT:  // decode found invalid UTF-8 byte
                 {
-                    std::stringstream ss;
-                    ss << std::setw(2) << std::uppercase << std::setfill('0') << std::hex << static_cast<int>(byte);
-                    JSON_THROW(type_error::create(316, "invalid UTF-8 byte at index " + std::to_string(i) + ": 0x" + ss.str()));
+                    std::string sn(3, '\0');
+                    snprintf(&sn[0], sn.size(), "%.2X", byte);
```

Comment:
How about changing this to:
`snprintf(&sn[0], sn.size(), "%.2X", static_cast<uint8_t>(s[i]));`

In order to avoid "‘%.2X’ directive output may be truncated writing between 2 and 8 bytes into a region of size 3 [-Wformat-truncation=]" (GCC 7.3.0).

Reply:
I don't see the difference.
`byte` is defined as
`const auto byte = static_cast<uint8_t>(s[i]);`
So it is 1 byte and the same, what you wrote.
Maybe a false positive in GCC?

Reply:
I suppose it might be a false positive since there is no difference.
Maybe someone with the same GCC version can confirm this warning.